ostd/src/arch/riscv/mm/mod.rs

```diff
@@ -102,13 +103,17 @@ pub(crate) fn tlb_flush_all_including_global() {
     riscv::asm::sfence_vma_all()
 }
 
+pub(crate) fn can_sync_dma() -> bool {
+    has_extensions(IsaExtensions::ZICBOM)
+}
+
 /// # Safety
 ///
-/// The caller must ensure that the virtual address range and DMA direction correspond correctly to
-/// a DMA region.
-pub(crate) unsafe fn sync_dma_range(range: Range<Vaddr>, direction: DmaDirection) {
-    if !has_extensions(IsaExtensions::ZICBOM) {
-        unimplemented!("DMA synchronization is unimplemented without ZICBOM extension")
+/// The caller must ensure that the virtual address range and DMA direction
+/// correspond correctly to a DMA region.
+pub(crate) unsafe fn sync_dma_range<D: DmaDirection>(range: Range<Vaddr>) {
+    if !can_sync_dma() {
+        panic!("DMA synchronization is unimplemented without ZICBOM extension");
     }
```

Comment:
I think this can be removed in preference of adding another safety comment that requires the existence of `ZICBOM` (or something similar to that: maybe requiring `can_sync_dma` is true for all the three architectures?).

We can still use `debug_assert!` on `can_sync_dma`, but I suppose the debug assertion isn't necessary since `cbo` instructions trigger an undefined opcode exception if the extension is absent (at least in practice, not sure if it's required by RISC-V spec).